### PR TITLE
fix: change public role like gamma procedure

### LINKED ISSUE #9807: Problems with Public role after updating
Hello! I've been running an older version of Superset without a problem, where I embed some dashboards on my website and through the Public user role permissions it is possible to see and use the dashboards. 

However, after updating to version 0.36 (using docker) I've been having problems providing public access to my dashboards. Without changing any user role settings, I get `Access Denied` messages when before I was able to access the dashboard. 

The only way I found around this is to set `PUBLIC_ROLE_LIKE_GAMMA = True` in the `superset_config.py` and then add the necessary permissions. However, every time I restart the server, it overwrites whatever permissions I've added and I have to add them again. If I put this setting back to False again, when I restart, the previous behavior comes up, where the public can't access the dashboard. 

It seems that for some reason Superset doesn't recognize the public user role unless if it is innitiated through the `PUBLIC_ROLE_LIKE_GAMMA` setting. 

Any ideas of what could be done to fix this? Thanks!

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Fixes #9807, by merging the `Public` role definition with pre-existing data access permissions (all other permissions not included on the template role are removed).

This PR also makes the definition of the `Public` role more configurable, for this `PUBLIC_ROLE_LIKE_GAMMA` is removed (breaking change) and a new key is introduced `PUBLIC_ROLE_LIKE` where a specific role can be set, like `Admin`, `Alpha`, `Gamma` or any other role preexisting on the DB or configured has a builtin role on the config itself.

To retain the same behaviour configure it the following way:

```
PUBLIC_ROLE_LIKE = "Gamma"
```

### ADDITIONAL INFORMATION

- [X] Has associated issue: #9807
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [X] Introduces new feature or API
- [X] Removes existing feature or API

### CODE REVIEW

File: superset/security/manager.py [hunk 1]
```diff
@@ -609,15 +619,67 @@ def sync_role_definitions(self) -> None:
         self.set_role("granter", self._is_granter_pvm)
         self.set_role("sql_lab", self._is_sql_lab_pvm)
 
-        if conf.get("PUBLIC_ROLE_LIKE_GAMMA", False):
-            self.set_role("Public", self._is_gamma_pvm)
+        if conf["PUBLIC_ROLE_LIKE"]:
+            self.copy_role(conf["PUBLIC_ROLE_LIKE"], self.auth_role_public, merge=True)
```
Comment (villebro) [inline superset/security/manager.py:623]: To be super accomodating, we could do something like
```python
if conf.get("PUBLIC_ROLE_LIKE_GAMMA", False):
    logger.warning("The config `PUBLIC_ROLE_LIKE_GAMMA` is deprecated and will be removed in Superset 1.0. Please use `PUBLIC_ROLE_LIKE ` instead.")
    self.copy_role("Gamma", self.auth_role_public, merge=True)
```

**Review by villebro — APPROVED** [review]
LGTM 🎉  One small convenience suggestion, although I'm not sure if we should be cluttering the codebase with these deprecation warnings, as `UPDATING.md` should be read by everyone anyway before updating.

File: superset/security/manager.py (hunk 1 above)
Comment (dpgaspar) [reply superset/security/manager.py:623]: Good idea

File: superset/config.py [hunk 2]
```diff
@@ -260,10 +260,10 @@ def _try_json_readsha(  # pylint: disable=unused-argument
 # ---------------------------------------------------
 # Roles config
 # ---------------------------------------------------
-# Grant public role the same set of permissions as for the GAMMA role.
+# Grant public role the same set of permissions as for a selected builtin role.
 # This is useful if one wants to enable anonymous users to view
 # dashboards. Explicit grant on specific datasets is still required.
-PUBLIC_ROLE_LIKE_GAMMA = False
+PUBLIC_ROLE_LIKE = "Gamma"
```
Comment (willbarrett) [inline superset/config.py:266]: This probably shouldn't default to "Gamma" - the old default behavior was to have a public role with no permissions.

Comment (mistercrunch) [reply superset/config.py:266]: +1, maybe `None` and handling that as noop

File: superset/config.py [hunk 3]
```diff
@@ -263,7 +263,7 @@ def _try_json_readsha(  # pylint: disable=unused-argument
 # Grant public role the same set of permissions as for a selected builtin role.
 # This is useful if one wants to enable anonymous users to view
 # dashboards. Explicit grant on specific datasets is still required.
-PUBLIC_ROLE_LIKE = "Gamma"
+PUBLIC_ROLE_LIKE = None
```
Comment (villebro) [inline superset/config.py:266]: I assume we should add the optional type, too:
```suggestion
PUBLIC_ROLE_LIKE: Optional[str] = None
```

**Review by willbarrett — APPROVED** [review]

**Comment by dpgaspar** [comment]
Added a very simple landing page for the public/anonymous users, since rendering the login page is not correct if the `Public` role is "activated". 
I think this is enough to establish a working workflow for the anonymous users, we can iterate on more pretty landing page on a following PR.

**Pull request merged**